site/docs/master/httpproxy.md

```diff
@@ -715,7 +715,65 @@ spec:
           port: 80
 ```
 
-#### ExternalName
+#### Path Rewriting
+
+HTTPProxy supports rewriting the HTTP request URL path prior to delivering the request to the service.
+Rewriting is performed after a routing decision has been made, and never changes the request destination.
+
+The `pathRewritePolicy` field specifies how the path prefix should be rewritten.
+The `replacePrefix` rewrite policy specifies a replacement string for a HTTP request path prefix match.
+When this field is present, the path prefix that the request matched is replaced by the text specified in the `replacement` field.
+The original path before rewriting will be placed into the into the `x-envoy-original-path` header.
+
+```yaml
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: rewrite-example
+  namespace: default
+spec:
+  virtualhost:
+    fqdn: rewrite.bar.com
+  routes:
+  - services:
+    - name: s1
+      port: 80
+    pathRewritePolicy:
+      replacePrefix:
+      - replacement: "/new/prefix"
+```
+
+The `replacePrefix` field accepts an array of possible replacements.
+When more than one `replacePrefix` array element is present, the `prefix` field can be used to disambiguate which replacement to apply.
+
+If no `prefix` field is present, the replacement is applied to all prefix matches made against the route.
```

Comment:
This part is confusing to me, if my path was `/`, it would replace `/foo`, `/foo/bar`, and `/foo/bar/zed/<anything>` with the specification in the `replacement` field. 

Is this what users would want since a longer path would always get replaced with the same resulting string?

Reply:
No, only "the path prefix that the request matched is replaced". I attempted to clarify.

Reply:
But if a path is delegated `/` then it will replace all paths passed since you didn't define the prefix to replace, wouldn't it?

Reply:
`/` is the prefix that would be replaced. The net result is that it would look like you prepended to the request path.